docs/cms/input-types.adoc

```diff
@@ -91,22 +91,56 @@ image::Content-selector-default.png[An input box with a dropdown options below i
   </config>
 </input>
 ----
-<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType Samples>>
+<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType>>
 <2> *allowPath* is used to limit the path of the content that may be selected using the <<allowPath Samples>>. By default, Content Selector
 only displays content from the same site.
 <3> *treeMode* toggles between tree and flat modes. Default value is `false` (flat mode). +
 image:Content-selector-tree.png[Input box with a dropdown with a tree structure, 350]
 <4> *hideToggleIcon* hides the toggle icon when set to `false`. +
 image:Content-selector-no-icon.png[Input box without an icon, 500]
 
-=== allowContentType Samples
+[#allowContentType]
+=== Allow content type pattern
+XP uses one of two modes when matching content type patterns:
 
+`LEGACY`:: this mode simulates the pattern matching functionality that was the default before XP 7.7.0. image:../images/xp-770.svg[XP 7.7.0,opts=inline] 
+`MATCH`:: this mode is the current default mode and has a few notable changes from `LEGACY` mode.
+
+The modes are explained in further detail below. You can configure which mode to use in  <<../deployment/config.adoc#admin, the admin config>>.
+==== Legacy +
+
+If the pattern value can be found in any part of the content type name, then it is considered a match.
+The pattern will be prefixed by the current application name unless:
+
+- the pattern uses the <application-name:content-name> format
+- the pattern includes one of `:` or `*`
+- the pattern starts with `${app}`.
+
+==== Match +
+A pattern will be treated as Java Regular Expression (with the exception to Special handling) and the whole content type name must match it to be accepted.
+The pattern will be prefixed by the current application name if it contains none of the following characters `\^$.|?*+()[{:`.
+
+==== Special handling
```

Comment:
My final suggestion 🙋🏾  I mentioned this briefly in a previous thread on this PR, but thought I'd put in a suggestion for it. This adds a link to the section about XP's differences and renames the title slightly.

I also reworded the sentence about characters that stop a pattern from being prefixed with the current application name for clarity.


```suggestion
The pattern will be treated as a Java Regular Expression (with <<#regex-differences, a few notable exceptions>>) and the whole content type name must match it to be accepted.
The pattern will be prefixed by the current application name if it does _not_ contain any of the following characters: `\^$.|?*+()[{:`.

[#regex-differences]
==== XP patterns: special cases

XP's patterns have a few cases where they differ from normal Java Regular Expression syntax:
```